packages/wouter-preact/types/index.d.ts

```diff
@@ -81,11 +81,12 @@ export type NavigationalProps<
 > = ({ to: Path; href?: never } | { href: Path; to?: never }) &
   HookNavigationOptions<H>;
 
-export type LinkProps<H extends BaseLocationHook = BrowserLocationHook> = Omit<
-  JSX.HTMLAttributes,
-  "href"
-> &
-  NavigationalProps<H>;
+type AsChildProps<RequiredProps, DefaultElementProps> =
+  | ({ asChild?: false } & RequiredProps & DefaultElementProps)
+  | ({ asChild: true; children: ComponentChildren } & RequiredProps);
+
+export type LinkProps<H extends BaseLocationHook = BrowserLocationHook> =
+  AsChildProps<NavigationalProps<H>, JSX.HTMLAttributes>;
```

Comment:
I have no experience with preact types, could you review it with more passion?

Reply:
I am not an expert either but could you try `JSX.IntrinsicElements["a"]`? This should give attributes tailored to `a` (mostly to get the proper type in event handlers).

Reply:
`JSX.IntrinsicElements["a"]` nope
it doesn't work

Reply:
Ok!